src/mlpack/core/cv/metrics/ssim_index_impl.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file ssim_index.hpp
+ * @author Utkarsh Rai
+ *
+ * The SSIM (Structural Similarity) metric.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_METRICS_SSIMINDEX_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_SSIMINDEX_IMPL_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace cv {
+
+template<typename DataType>
+double SSIMIndex::Evaluate(DataType const& image,
+                           DataType const& reference)
+{
+  if (arma::size(reference) != arma::size(image))
+  {
+    std::ostringstream oss;
+    oss << "SSIMIndex::Evaluate(): size of reference ("
+      <<reference.n_cols<<"X"<<reference.n_rows
+      <<") does not match size of image ("
+      <<image.n_cols<<"X"<<image.n_rows<<")!"
+      <<std::endl;
+    throw std::invalid_argument(oss.str());
+  }
+  // Instantitating a Gaussian Kernel with bandwidth 1.5
+  kernel::GaussianKernel gaussianKernel = kernel::GaussianKernel(1.5);
+  
+  // Gaussian Window of size 11X11.
+  arma::mat gaussianWindow(11, 11);
+  for ( double row = 0; row < 11; row+=1)
+  {
+    for ( double column = 0; column<11; column+=1)
+    {
+      gaussianWindow(row, column) = gaussianKernel.Evaluate(
+                                    arma::linspace(-6, row - 5, 1),
+                                    arma::linspace(-6, column - 5, 1));
+    }
+  }
+  // Calculate local mean of reference
```

Comment:
```suggestion
  // Calculate local mean of reference.
```
Could you also please add stops at the end of the comments elsewhere too. Thanks.